template/ios/Podfile

```diff
@@ -33,5 +33,7 @@ target 'HelloWorld' do
   post_install do |installer|
     react_native_post_install(installer)
     __apply_Xcode_12_5_M1_post_install_workaround(installer)
+    # Enable the next line in order to apply patches necessary for Mac Catalyst builds
+    #__apply_mac_catalyst_patches(installer)
```

Comment:
I'll move the invocation of `__apply_mac_catalyst_patches(installer)` inside the `react_native_post_install`, actually removing the `__apply_mac_catalyst_patches`. Given that not all the projects need these, we can update the `react_native_post_install` function to accept a boolean `apply_catalyst_patch` with a default value of `false`.

The rationale here is that:
1. it helps discoverability. At the end of the refactoring effort, the `react_native_pods` script will have only two function: `use_react_native` and `react_native_post_install`. These will be documented and people will know about the existence of this flag. Otherwise, they have to read the whole script file to learn bout its existence.
2. it makes it easier for us to maintain the script. When, in the future, we will have to remove it, we could just delete all the code, and deprecate the flag. No other change will be required by the apps for a while, until we actually remove the flag.

Note that this change may foce you to also update the RNTester app.

How do you feel about this? Does it make sense?